### DISCUSSION TITLE
Oauth2 Redirect Behind Nginx | Don't Expose Admin Dashboard

### DISCUSSION THREAD

**Discussion by HenryHuevo** [pocketbase/pocketbase#3225] [Q&A] [answered]
Hey there, apologies if this was already asked.  Basically I use svelte for my frontend and pocketbase for my backend (note **_svelte_**, not svelte**_kit_**).  My users never need to access the pocketbase UI, so I only proxy /api to localhost:8090.  The downside to this is while Pocketbase successfully completes oauth and logs the user in and sets the proper local storage, the pop-up window does not automatically close itself, instead resulting in a 404.

I use this example from the documentation:
const authData = await pb.collection('users').authWithOAuth2({ provider: 'google' });

If I pass through all traffic on /_/, it obviously works, but I don't want the Admin Dashboard available publicly.  Not that there is anything wrong with this, I am just of the mindset "If it doesn't need to be public, don't make it public".  Will I have to try to use a manual code exchange?  I suppose I am just trying to dodge that responsibility to keep my implementation as simple as possible.  Plus, with the svelte rendering the rest of the site content statically, I am unsure if I can add a custom route with it, but that is a Svelte problem, not a pocketbase problem.

I have a similar project that uses `authWithPassword`, and I do not have this problem (No popup == No problem).

Alternatively, is there a way I can tweak the behavior of the pop-up window to auto-close on a certain response (ex: 404)?

Thanks

**Comment by ganigeorgiev** [comment, marked as answer]
If you need a custom OAuth2 redirect handling you'll have to implement the manual code exchange flow.

But keep in mind that hiding the Admin UI doesn't really help with the security. The Admin UI is a plain client-side application that just makes request to the api.

If you want to prevent external admin authentication you can define a nginx rule that will disallow `/api/admins/*` requests (_or allow them only for your IP_). This way the builtin OAuth2 redirect page can remain accessible and even though the user will see the admin login form when they access `/_/`, the auth request will fail.

**Reply by HenryHuevo** [reply]
Appreciate the fast response!  I was under the impression that the `/_/` route was a different set of requests and was unaware that it made requests to `/api`.  Awesome!  I think I'll give the nginx disallow a try to neuter the admin dashboard functionality.

Thanks a lot!
